trace/trace.go

```diff
@@ -108,9 +98,22 @@ func (tr *tracePlugin) AfterExecution(ctx context.Context, request json.RawMessa
 	tr.RootSpan.SetTag(constants.AwsLambdaLogGroupName, application.LogGroupName)
 	tr.RootSpan.SetTag(constants.AwsLambdaLogStreamName, application.LogStreamName)
 	tr.RootSpan.SetTag(constants.AwsLambdaInvocationColdStart, invocationCount == 1)
-	tr.RootSpan.SetTag(constants.AwsLambdaInvocationTimeout, utils.IsTimeout(err))
 	tr.RootSpan.SetTag(constants.AwsLambdaInvocationRequestId, application.GetAwsRequestID(ctx))
 
+	tracer.OnSpanStarted(tr.RootSpan)
+
+	return ctx
+}
+
+// AfterExecution executes the necessary tasks after the invocation
+func (tr *tracePlugin) AfterExecution(ctx context.Context, request json.RawMessage, response interface{}, err interface{}) ([]plugin.MonitoringDataWrapper, context.Context) {
+	finishTime, ctx := plugin.EndTimeFromContext(ctx)
+	tr.Data.FinishTime = finishTime
+	tr.Data.Duration = tr.Data.FinishTime - tr.Data.StartTime
+	tr.RootSpan.FinishWithOptions(opentracing.FinishOptions{FinishTime: utils.MsToTime(finishTime)})
```

Comment:
`onFinished` hook is not called for the root span since it's called in the `Finish` method [here](https://github.com/thundra-io/thundra-lambda-agent-go/blob/463a14af3ec85ce465a11c136f3179599ffba9ce/tracer/span.go#L24). So the span listeners that are set to run on finished don't have any effect on the root span currently.